internal/artifacts/module_arch_test.go

```diff
@@ -859,34 +847,29 @@ module-types:
 			mtaObj = getMtaObj("mtahtml5", "mtaWithWrongBuildRequirements.yaml")
 		})
 
-		Describe("scenarios based on mtaWithWrongBuildRequirements.yaml", func() {
-			It("sanity", func() {
-				modules, err := getRequiredModules(mtaObj, "m1")
-				Ω(err).Should(Succeed())
-				Ω(modules).Should(Equal([]string{"m4", "m3", "m2"}))
-			})
-			DescribeTable("failures", func(targetModule string) {
-				_, err := getRequiredModules(mtaObj, targetModule)
-				Ω(err).Should(HaveOccurred())
-			},
-				Entry("fails on none existing target module", "m5"),
-				Entry("fails on none existing module required by the target module", "n1"),
-				Entry("fails on none existing module required by the module that is required by the target module", "n2"))
+		It("sanity", func() {
+			modules, err := getRequiredModules(mtaObj, "m1")
+			Ω(err).Should(Succeed())
+			Ω(modules).Should(Equal([]string{"m4", "m3", "m2"}))
 		})
+		DescribeTable("failures", func(targetModule string) {
+			_, err := getRequiredModules(mtaObj, targetModule)
+			Ω(err).Should(HaveOccurred())
+		},
+			Entry("fails on none existing target module", "m5"),
+			Entry("fails on none existing module required by the target module", "n1"),
+			Entry("fails on none existing module required by the module that is required by the target module", "n2"))
 	})
 
 	Describe("sortSelectedModules", func() {
 		It("sanity", func() {
 			mtaObj := getMtaObj("mtahtml5", "mtaWithBuildRequirements.yaml")
-			modules, err := sortSelectedModules(mtaObj, []string{"n1", "m1"})
+			allModulesSorted, err := buildops.GetModulesNames(mtaObj)
 			Ω(err).Should(Succeed())
-			Ω(modules).Should(Equal([]string{"m1", "n1"}))
-		})
-
-		It("fails on the mta yaml problems", func() {
-			mtaObj := getMtaObj("mtahtml5", "mtaWithWrongBuildRequirements.yaml")
-			_, err := sortSelectedModules(mtaObj, []string{"m1", "n1"})
-			Ω(err).Should(HaveOccurred())
+			selectedModules := []string{"n1", "m1"}
+			selectedModulesMap := convertListToMap(selectedModules)
+			selectedModulesSorted := sortSelectedModules(allModulesSorted, selectedModulesMap)
```

Comment:
This looks like too much work for a test. Maybe create the map from the beginning instead of using an array or have `sortSelectedModules` receive an array and convert it to a map internally.